src/client.ts

```diff
@@ -163,9 +168,18 @@ export class Connection {
   public async close(
     params: { closingCode: number; closingReason: string } = { closingCode: 0, closingReason: "" }
   ): Promise<void> {
-    this.logger.info(`Closing connection ...`)
+    this.logger.info(`Closing client...`)
+    if (this.producerCounts()) {
```

Comment:
is it needed to check for the number of producers/consumers before closing them?

Reply:
It's mostly in order to avoid confusing logs whenever we close the producers/consumers own Clients, so that they don't print "Closing consumers..." as well